draft-ietf-ipsecme-add-ike.xml

```diff
@@ -391,36 +398,38 @@
     </section>
 
     <section anchor="protocol" title="IKEv2 Protocol Exchange">
-      <t>This section describes how an initiator can be configured with an
-      encrypted DNS resolver using IKEv2.</t>
+      <t>This section describes how the attributes defined in <xref target="RI"></xref>
+      are used to configure an IKEv2 initiator with one or more encrypted DNS resolvers.</t>
 
-      <t>Initiators indicate the support of an encrypted DNS in the
-      CFG_REQUEST payloads by including one or two ENCDNS_IP* attributes,
-      while responders supply the encrypted DNS configuration in the CFG_REPLY
-      payloads. Concretely:</t>
+      <t>Initiators first indicate support for encrypted DNS by including ENCDNS_IP*
+      attributes in their CFG_REQUEST payloads.
+      Responders supply encrypted DNS configuration by including ENCDNS_IP* attributes
+      in their CFG_REPLY payloads. Concretely:</t>
 
       <t><list style="empty">
-          <t>If the initiator supports encrypted DNS, it includes one or two
-          ENCDNS_IP* attributes in the CFG_REQUEST. For each IP address family
-          the initiator MUST include exactly one attribute with the Length
-          field set to 0 if no specific DNS resolver is requested. The
-          initiator MAY include the ENCDNS_DIGEST_INFO attribute with a list
-          of hash algorithms that are supported by the encrypted DNS
-          client.</t>
+          <t>If the initiator supports encrypted DNS, it includes either or both
+          of the ENCDNS_IP4 and ENCDNS_IP6 attributes in its CFG_REQUEST. If
+          the initiator does not want to request specific DNS resolvers, it sets
+          the Length field to 0 for the attribute. The initiator MUST NOT send
```

Comment:
I don't think this MUST NOT is necessary (at least in RFC 2119 language). If for some reason multiple empty attributes are sent, with the current language the responder has to treat this as a fatal error (violation of MUST NOT) and most probably tear down the connection. In fact it's harmless (except for waste of bandwidth).

Reply:
The responder may typically ignore the violation but if the client tries to takes advantage of it to waste the cycles of the responder (e.g., repeating the attribute several hundred times; the resolver may want to treat it is a fatal error). we can probably say: 

If the receiver gets repeated attributes, it SHOULD ignore it.

Reply:
I can live with SHOULD :-)